packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -19,6 +19,13 @@ const double _kHandlesPadding = 18.0;
 const double _kToolbarScreenPadding = 8.0;
 const double _kToolbarHeight = 36.0;
 
+// If the distance from the top is less than a certain value,
+// the toolbar should be displayed below the input box.
+// If don't do this, won't be able to properly display and
+// interact toolbar at the top of some phones, such as iPhone X.
```

Comment:
Some grammar issue with this sentence `properly display and interact toolbar`.

Reply:
I'm not sure whether it's so much that we need a specific value of a padding to avoid or whether we should just make sure we account for MediaQuery.of(context).padding in general when we construct the _TextSelectionToolbarLayout (and use it in the calculation). 

For instance, without the notch (and consequently the media query padding), iOS seems to put the toolbar right up against the edge of the screen.
![screenshot-2019-04-10-22-35-16](https://user-images.githubusercontent.com/156888/55918178-f7b45480-5ba6-11e9-85b0-2a5ade1fa43a.png)

Reply:
Thanks Reply.

The constant has now been removed, and the toolbar can dynamically adjust the display based on the actual position of the input box.

iPhone 6SP (Flutter, this branch):
![](https://raw.githubusercontent.com/jiisd/images_temp/master/fix_flutter_issues_29808_2.png)
iPhone XR (Flutter, this branch):
![](https://raw.githubusercontent.com/jiisd/images_temp/master/fix_flutter_issues_29808_3.png)